orc8r/protos/service_registry.proto

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+syntax = "proto3";
+package magma.orc8r;
+
+import "orc8r/protos/common.proto";
+
+option go_package = "magma/orc8r/lib/go/protos";
+
+
+// --------------------------------------------------------------------------
+// The ServiceRegistry interface implements a set of functionalities that can
+// be used for service discovery.
+// --------------------------------------------------------------------------
+service ServiceRegistry {
```

Comment:
Preference for the Uber style [0] where every request/response message is named directly after the RPC -- e.g. `rpc ListAllServices (ListAllServicesRequest) returns (ListAllServicesResponse) {}`

[0] https://github.com/uber/prototool/blob/dev/style/README.md#servicesrpcs